Add Connectors and ML updates for 2.9 (#4554)
opensearch-project/documentation-website: _ml-commons-plugin/ml-framework.md

---
layout: default
title: ML framework 
has_children: true
nav_order: 50
redirect_from:
   - /ml-commons-plugin/model-serving-framework/
---

> ML Framework was taken out of experimental status and released to General Availability in OpenSearch 2.9.  

Reviewer: ```suggestion
ML Framework was taken out of experimental status and released as Generally Available in OpenSearch 2.9.  
```

> {: .warning}

Author: ```suggestion
{: .note}
```

> # ML framework

Reviewer: ```suggestion
# ML Framework
```

> ML Commons allows you to serve custom models and use those models to make inferences through OpenSearch's machine learning framework (ML framework) For those who want to run their PyTorch deep learning model inside an OpenSearch cluster, you can upload and run that model with the ML Commons REST API.

Reviewer: ```suggestion
ML Commons allows you to serve custom models and use those models to make inferences through the OpenSearch Machine Learning (ML) Framework. For those who want to run their PyTorch deep learning model inside an OpenSearch cluster, you can upload and run that model with the ML Commons REST API.
```

> This page outlines the steps required to upload a custom model and run it with the ML framework.

Reviewer: ```suggestion
This page outlines the steps required to upload a custom model and run it with the ML Framework.
```

### Model support

> As of OpenSearch 2.6, the ML framework supports text embedding models.

Reviewer: ```suggestion
As of OpenSearch 2.6, the ML Framework supports text-embedding models.
```

## GPU acceleration

> To achieve better performance within the ML framework, you can take advantage of GPU acceleration on your ML node. For more information, see [GPU acceleration]({{site.url}}{{site.baseurl}}/ml-commons-plugin/gpu-acceleration/).

Reviewer: ```suggestion
To achieve better performance within the ML Framework, you can take advantage of GPU acceleration on your ML node. For more information, see [GPU acceleration]({{site.url}}{{site.baseurl}}/ml-commons-plugin/gpu-acceleration/).
```
